mu4/scenes/inspector/iinspectoradapter.h

```diff
@@ -0,0 +1,76 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#ifndef IINSPECTORADAPTER_H
+#define IINSPECTORADAPTER_H
+
+#include <QVariant>
+
+#include "modularity/imoduleexport.h"
+
+#include "libmscore/style.h"
+
+namespace Ms {
+class Score;
+}
+
+namespace mu {
+namespace scene {
+namespace inspector {
+class IInspectorAdapter : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(IInspectorAdapter)
+public:
+    virtual ~IInspectorAdapter() = default;
+
+    virtual bool isNotationExisting() const = 0;
+
+    // notation commands
+    virtual void beginCommand() = 0;
+    virtual void endCommand() = 0;
+
+    // notation styling
+    virtual void updateStyleValue(const Ms::Sid& styleId, const QVariant& newValue) = 0;
+    virtual QVariant styleValue(const Ms::Sid& styleId) = 0;
+
+    // dialogs
+    virtual void showSpecialCharactersDialog() = 0;
+    virtual void showStaffTextPropertiesDialog() = 0;
+    virtual void showPageSettingsDialog() = 0;
+    virtual void showStyleSettingsDialog() = 0;
+    virtual void showTimeSignaturePropertiesDialog() = 0;
+    virtual void showArticulationPropertiesDialog() = 0;
+    virtual void showGridConfigurationDialog() = 0;
```

Comment:
It would be great to call all these dialogues on URI through interactive. In MU4, you can prescribe these URI, but not implement it in this PR. And for MU3, make your own implementation of interactive, in which there would be a check for URI and show the corresponding dialogue.

Reply:
Ye, we can do it later. The PR is already gigantic, so I focused on the transfer.